subprojects/docs/src/samples/user-input/pass-credentials-to-external-tool-via-stdin/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,13 @@
+val login = tasks.register<Exec>("login") {
+    val username: String by project
+    val password: String by project
+    standardInput = java.io.ByteArrayInputStream("$username\n$password".toByteArray())
```

Comment:
I think this fails when username/password aren't provided and you run anything that realizes this task. e.g., `gradle tasks` would trigger a configuration failure because these properties aren't set.

This may need to be a little fancier so that we only unwind the inputs when they're asked for (vs at configuration time).  This sort of thing would break otherwise:
```
tasks.register<Exec>("login") {
    ... (what's here)
}

// Only bob runs this build
project.username = "bob"
```

The build can't contribute to the value of username/password because it's read eagerly.

Reply:
I switched to using the `ProviderFactory` to obtain property providers and added a check to assert the values are present in `doFirst` so that provider values are evaluated and validated only when the task is executed.